app/connectors/VatConnector.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import javax.inject.Inject
+import model.Vrn
+import model.vat.{CalendarData, VatDesignatoryDetailsAddress}
+import play.api.{Configuration, Logger}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class VatConnector @Inject() (
+    servicesConfig: ServicesConfig,
+    httpClient:     HttpClient,
+    configuration:  Configuration)
+  (implicit ec: ExecutionContext) {
+
+  lazy private val vatUrl: String = servicesConfig.baseUrl("vat")
+
+  def calendar(vrn: Vrn)(implicit hc: HeaderCarrier): Future[Option[CalendarData]] = {
```

Comment:
You throw exception later in view if the calendar is `None`. Maybe it's better to change the signature to `Future[CalendarData]`?

Reply:
No, we don't throw exception .. as we may not have a vat return. you are thinking of designatoryDetails, will update the same ;)